File: ninchatsdk/src/main/java/com/ninchat/sdk/adapters/holders/formview/NinchatTextViewHolder.java
package com.ninchat.sdk.adapters.holders.formview;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;
import com.ninchat.sdk.R;
import com.ninchat.sdk.utils.misc.Misc;

import org.json.JSONObject;

import static com.ninchat.sdk.helper.questionnaire.NinchatQuestionnaireItemGetter.*;
import static com.ninchat.sdk.helper.questionnaire.NinchatQuestionnaireMiscUtil.setAnimation;

public class NinchatTextViewHolder extends RecyclerView.ViewHolder {
    private TextView mTextView;

    public NinchatTextViewHolder(@NonNull View itemView,
                                 JSONObject questionnaireElement,
                                 boolean isFormLikeQuestionnaire, int position) {
        super(itemView);
        mTextView = itemView.findViewById(R.id.text_view_content);
        bind(questionnaireElement, isFormLikeQuestionnaire, position, false);
    }

    public void bind(JSONObject questionnaireElement, boolean isFormLikeQuestionnaire, int position, boolean isUpdate) {
        if (questionnaireElement == null) {
            return;
        }
        String labelText = getLabel(questionnaireElement);
        if (isFormLikeQuestionnaire) {
            itemView.setBackground(
                    ContextCompat.getDrawable(itemView.getContext(), R.drawable.ninchat_chat_form_questionnaire_background)
            );
        }

        // there might be some images images
        mTextView.setText(Misc.toRichText(labelText, mTextView));
        if (!isUpdate)
            setAnimation(itemView, position, position != 0);
    }
}

File: ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchattext/view/NinchatTextViewHolder.kt
package com.ninchat.sdk.ninchatquestionnaire.ninchattext.view

import android.view.View
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.ninchat.sdk.R
import com.ninchat.sdk.ninchatquestionnaire.ninchattext.presenter.INinchatTextViewPresenter
import com.ninchat.sdk.ninchatquestionnaire.ninchattext.presenter.NinchatTextViewPresenter
import com.ninchat.sdk.utils.misc.Misc
import kotlinx.android.synthetic.main.text_view.view.*
import org.json.JSONObject

class NinchatTextViewHolder(
        itemView: View,
        jsonObject: JSONObject?,
        isFormLikeQuestionnaire: Boolean = true,
) : RecyclerView.ViewHolder(itemView), INinchatTextViewPresenter {

    private val ninchatTextViewPresenter = NinchatTextViewPresenter(
            jsonObject = jsonObject,
            isFormLikeQuestionnaire = isFormLikeQuestionnaire,
            iPresenter = this
    )

    fun update(jsonObject: JSONObject?, isFormLikeQuestionnaire: Boolean = true) {
        ninchatTextViewPresenter.renderCurrentView()
    }

    override fun onUpdateConversationView(label: String?) {
        itemView.text_view_content.text = Misc.toRichText(label, itemView.text_view_content)
    }

    override fun onUpdateFormView(label: String?) {
        val background = R.drawable.ninchat_chat_questionnaire_background
        itemView.text_view_content.text = Misc.toRichText(label, itemView.text_view_content)
        itemView.background = ContextCompat.getDrawable(itemView.context, background)
    }
}